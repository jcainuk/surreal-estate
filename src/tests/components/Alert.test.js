import React from "react";
import { render, screen } from "@testing-library/react";
import Alert from "../../components/Alert";

describe("Alert", () => {
  test("it displays an error message", () => {
    const { asFragment } = render(
      <Alert
        message="Server error. Please contact customer support."
        isSuccess={false}
      />
    );
    expect(
      screen.getByText("Server error. Please contact customer support.")
    ).toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot();
  });

  test("it displays a success message", () => {
    const { asFragment } = render(<Alert message="Property added" isSuccess />);
    expect(screen.getByText("Property added")).toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot();
  });

  test("it does not render when there is no alert message", () => {
    const { asFragment } = render(<Alert message="" isSuccess={false} />);
    const alert = asFragment();
    expect(alert).toMatchSnapshot();
  });
});
