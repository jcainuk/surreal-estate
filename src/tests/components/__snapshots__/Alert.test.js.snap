// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert it displays a success message 1`] = `
<DocumentFragment>
  <div
    class="Alert alert-error"
  >
    Property added
  </div>
</DocumentFragment>
`;

exports[`Alert it displays an error message 1`] = `
<DocumentFragment>
  <div
    class="Alert alert-error"
  >
    Server error. Please contact customer support.
  </div>
</DocumentFragment>
`;

exports[`Alert it does not render when there is no alert message 1`] = `<DocumentFragment />`;
